From 89e926a5e57c708ddafa66c1ad7ad6bfbb42124a Mon Sep 17 00:00:00 2001
From: He Zhe <zhe.he@windriver.com>
Date: Tue, 23 Dec 2014 16:30:30 +0800
Subject: [PATCH 03/10] fs/Kconfig: make MAX_NESTED_LINKS configurable

Specifies the nesting limit on symbolic links. But keep in mind that
setting it too high could lead to kernel stack overflows.

Signed-off-by: Yongli he   <yongli.he@windriver.com>
Signed-off-by: He Zhe <zhe.he@windriver.com>
---
 fs/Kconfig            |    8 ++++++++
 include/linux/namei.h |    4 ++++
 2 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/fs/Kconfig b/fs/Kconfig
index 9d207b1..2ea7046 100644
--- a/fs/Kconfig
+++ b/fs/Kconfig
@@ -62,6 +62,14 @@ config FS_POSIX_ACL
 config EXPORTFS
 	tristate
 
+config MAX_NESTED_LINKS
+	int "Symblic link nest limit"
+	default 8
+	help
+	  Specifies the nesting limit on symbolic links.  You may increase
+	  this value, but setting it too high could lead to kernel stack
+	  overflows.
+
 config FILE_LOCKING
 	bool "Enable POSIX file locking API" if EXPERT
 	default y
diff --git a/include/linux/namei.h b/include/linux/namei.h
index c899077..4035277 100644
--- a/include/linux/namei.h
+++ b/include/linux/namei.h
@@ -9,7 +9,11 @@
 struct vfsmount;
 struct nameidata;
 
+#ifdef CONFIG_MAX_NESTED_LINKS
+enum { MAX_NESTED_LINKS = CONFIG_MAX_NESTED_LINKS };
+#else
 enum { MAX_NESTED_LINKS = 8 };
+#endif
 
 /*
  * Type of the last component on LOOKUP_PARENT
-- 
1.7.5.4

